main.go

```diff
@@ -132,63 +132,98 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
-	milestones := []milestoneAPI{}
-
+func getActiveMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
 	strURL := []string{baseURL, projectID, "/milestones?state=active&per_page=100"}
 	URL := strings.Join(strURL, "")
+	milestones := map[string]string{}
 
-	m, linkHeader, err := callServer(URL, token, milestones)
+	m, err := gitlabGet(URL, token)
 	if err != nil {
-		return m, err
+		return milestones, err
 	}
 
-	for _, milestone := range milestones {
-		m[milestone.Title] = milestone.DueDate
+	for _, milestone := range m {
+		milestones[milestone.Title] = milestone.DueDate
 	}
 
-	parsedHeader := link.Parse(linkHeader)
-	for _, elem := range parsedHeader {
-		if elem.Rel == "next" {
-			m, _, err := callServer(URL, token, milestones)
-			if err != nil {
-				return m, err
-			}
-			for _, milestone := range milestones {
-				m[milestone.Title] = milestone.DueDate
-			}
-		}
+	return milestones, nil
+}
+
+// Get and return inactive milestones
+func getInactiveMilestones(baseURL string, token string, projectID string) (map[string]int, error) {
+	strURL := []string{baseURL, projectID, "/milestones?state=closed&per_page=100"}
+	URL := strings.Join(strURL, "")
+	milestones := map[string]int{}
+
+	m, err := gitlabGet(URL, token)
+	if err != nil {
+		return milestones, err
 	}
 
-	return m, nil
+	for _, milestone := range m {
+		milestones[milestone.Title] = milestone.ID
+	}
+
+	return milestones, nil
 }
 
-func callServer(URL string, token string, milestones []milestoneAPI) (map[string]string, string, error) {
-	m := map[string]string{}
+func gitlabGet(URL string, token string) ([]milestoneAPI, error) {
```

Comment:
Not sure on the naming of that function, as it's not very explicit what it does (I'd suggest `getMilestones`, as a general function that can be called by `getInactiveMilestones` and `getActiveMilestones`, which in the end do the same thing, only they specify different filters for the request)

Which brings me too:

`getInactive` and `getActive` look very similar, the only thing where they differ being the options passed to the GET request. Maybe all the common logic could be put into `getMilestones`, and the other two functions would just be something like
```go
func getInactiveMilestones(baseURL, token, projectID string) ([]milestoneAPI, error) {
    return getMilestones(baseURL, token, projectID, []string{"state", "closed", "per_page", "100"})
}
```

(simplified)

Reply:
Also maybe instead of mapping title to ID or DueDate only, we could just return the `[]milestoneAPI` array, and then map the data that we're interested in outside of the function, depending on what we want to do with it. That would make these functions more reusable

Reply:
What about:
```
func getInactiveMilestones(baseURL, token, projectID string) ([]milestoneAPI, error) {
    return getMilestones(baseURL, token, projectID, state)
}
```
And in addition to that we create a map with a simplified milestone.
```
type milestone struct {
	DueDate string
	ID      string
}
func main() {
	var m milestone
	milestones := map[string]milestone{}
	m.DueDate = "2018-01-01"
	m.ID = "24"
	milestones["2018-01-01"] = m
	fmt.Println(milestones)
}
```

That would give us the ability to querry easily for title and extract the DueDate and the ID.
DueDate would be usable for not yet created milestones and ID used for reactivating milestones. There is actually no need to have DueDate set on reactivation as only changed fields should be overwritten.

Reply:
Also I agree that getMilestones should return []milestoneAPI first and then this should be formed into the above mentioned map and struct. So it can be used for creation and updating of the milestones.